pkg/server/list_users.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/openfga/openfga/pkg/middleware/validator"
+	"github.com/openfga/openfga/pkg/telemetry"
+
+	"github.com/openfga/openfga/pkg/server/commands/listusers"
+	serverErrors "github.com/openfga/openfga/pkg/server/errors"
+	"github.com/openfga/openfga/pkg/typesystem"
+)
+
+// ListUsers returns all subjects (users) of a specified terminal type
+// that are relate via specific relation to a specific object.
+func (s *Server) ListUsers(
+	ctx context.Context,
+	req *openfgav1.ListUsersRequest,
+) (*openfgav1.ListUsersResponse, error) {
+	start := time.Now()
+	ctx, span := tracer.Start(ctx, "ListUsers", trace.WithAttributes(
+		attribute.String("store_id", req.GetStoreId()),
+		attribute.String("object", fmt.Sprintf("%s:%s", req.GetObject().GetType(), req.GetObject().GetId())),
+		attribute.String("relation", req.GetRelation()),
+		attribute.String("user_filters", userFiltersToString(req.GetUserFilters())),
+	))
+	defer span.End()
+	if !s.IsExperimentallyEnabled(ExperimentalEnableListUsers) {
+		return nil, status.Error(codes.Unimplemented, "ListUsers is not enabled. It can be enabled for experimental use by passing the `--experimentals enable-list-users` configuration option when running OpenFGA server")
+	}
+
+	if !validator.RequestIsValidatedFromContext(ctx) {
+		if err := req.Validate(); err != nil {
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
+	}
+
+	const methodName = "listusers"
+
+	typesys, err := s.resolveTypesystem(ctx, req.GetStoreId(), req.GetAuthorizationModelId())
+	if err != nil {
+		telemetry.TraceError(span, err)
+		return nil, err
+	}
+
+	err = listusers.ValidateListUsersRequest(ctx, req, typesys)
+	if err != nil {
+		telemetry.TraceError(span, err)
+		return nil, err
+	}
+
+	ctx = typesystem.ContextWithTypesystem(ctx, typesys)
+
+	listUsersQuery := listusers.NewListUsersQuery(s.datastore,
+		listusers.WithResolveNodeLimit(s.resolveNodeLimit),
+		listusers.WithResolveNodeBreadthLimit(s.resolveNodeBreadthLimit),
+		listusers.WithListUsersQueryLogger(s.logger),
+		listusers.WithListUsersMaxResults(s.listUsersMaxResults),
+		listusers.WithListUsersDeadline(s.listUsersDeadline),
+		listusers.WithListUsersMaxConcurrentReads(s.maxConcurrentReadsForListUsers),
+	)
+
+	resp, err := listUsersQuery.ListUsers(ctx, req)
+	if err != nil {
+		telemetry.TraceError(span, err)
+
+		switch {
+		case errors.Is(err, graph.ErrResolutionDepthExceeded):
+			return nil, serverErrors.AuthorizationModelResolutionTooComplex
+		case errors.Is(err, condition.ErrEvaluationFailed):
+			return nil, serverErrors.ValidationError(err)
+		default:
+			return nil, serverErrors.HandleError("", err)
+		}
+	}
+
+	datastoreQueryCount := float64(resp.Metadata.DatastoreQueryCount)
+
+	grpc_ctxtags.Extract(ctx).Set(datastoreQueryCountHistogramName, datastoreQueryCount)
+	span.SetAttributes(attribute.Float64(datastoreQueryCountHistogramName, datastoreQueryCount))
+	datastoreQueryCountHistogram.WithLabelValues(
+		s.serviceName,
+		"list-users",
```

Comment:
```suggestion
		methodName,
```
Should be consistent with the others.